src/main/java/org/folio/innreach/domain/service/impl/KafkaEventProcessorServiceImpl.java

```diff
@@ -24,4 +32,23 @@ public <T> void process(T event, Consumer<T> eventProcessor, String tenant) {
       log.warn("Unable to save kafka event into outbox table ", ex);
     }
   }
+
+  @Override
+  public <T> void process(List<DomainEvent<T>> events, BiConsumer<List<DomainEvent<T>>, String> eventProcessor) {
+    try {
+      Map<String, List<DomainEvent<T>>> tenantMap = events.stream().collect(Collectors.groupingBy(DomainEvent::getTenant));
+      tenantMap.forEach((tenant, eventList) -> {
+        log.debug("tenant {}", tenant);
+        log.debug("eventList for tenant {}", eventList);
+        if (innReachTenants.contains(tenant)) {
+          executionService.runTenantScoped(tenant,
+            () -> eventProcessor.accept(eventList, tenant));
+        } else {
+          log.warn("Ignoring event of unknown tenant {}", tenant);
+        }
+      });
+    } catch (Exception ex) {
+      log.warn("Unable to save kafka event into outbox table ", ex);
```

Comment:
Should this be an `error` log and maybe also add details or event itself to the message.

Reply:
Yes, changed to error